frame-metadata/src/lib.rs

```diff
@@ -20,14 +20,32 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 #![warn(missing_docs)]
 
+#[cfg(all(
+	feature = "scale_info",
+	any(
+		feature = "v13",
+		feature = "v12",
+		feature = "v11",
+		feature = "v10",
+		feature = "v9",
+		feature = "v8",
+		feature = "legacy"
+	)
+))]
+compile_error!("Metadata prior to version v14 don't support scale-info");
```

Comment:
I did not notice this before, but this compile error is actually not correct because you can enable multiple different metadata versions at the same time. Hence, we should permit having scale-info and the old metadata too.

We could just add a comment in the toml above the feature flag:

```
# Derives Decode, Serialize and Debug for the newest metadata version
```

Reply:
old metadata can only compile with std, not with scale-info. I don't know how you can enable different version in no_std, as only v14 is supporting it.